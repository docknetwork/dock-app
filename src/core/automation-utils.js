export function addTestId(testId) {
  return {
    testID: testId,
    accessibilityLabel: testId,
  };
}
