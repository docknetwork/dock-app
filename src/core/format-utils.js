export function formatCurrency(value, currency = 'USD') {
  var formatter = new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: currency,

    minimumFractionDigits: 2,
    maximumFractionDigits: 10,
  });

  return formatter.format(value);
}

export function formatAddress(value, size = 19) {
  if (!value || size > value.length) {
    return value;
  }

  const offset = size / 2;
  return `${value.substring(0, offset)}...${value.substring(
    value.length - offset,
  )}`;
}
