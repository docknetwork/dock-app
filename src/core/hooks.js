import {useEffect} from 'react';
import {BackHandler} from 'react-native';

export function useBackHandler(callback) {
  useEffect(() => {
    const backHandler = BackHandler.addEventListener(
      'hardwareBackPress',
      () => {
        console.log('stuff');

        return true;
      },
    );

    return () => backHandler.remove();
  }, []);
}
