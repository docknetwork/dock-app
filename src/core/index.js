/**
 * JSON.stringify with error handler
 *
 * @param value
 * @returns {string}
 */
export function getJsonOrError(value) {
  try {
    return JSON.stringify(value);
  } catch (err) {
    return err;
  }
}
