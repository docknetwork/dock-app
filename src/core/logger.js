const winston = require('winston');

export const Logger = winston.createLogger({
  format: winston.format.simple(),
  levels: ['debug'],
  transports: [new winston.transports.Console()],
});
