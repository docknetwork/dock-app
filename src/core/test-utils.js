import {NativeBaseProvider} from 'native-base';
import React from 'react';

const inset = {
  frame: {x: 0, y: 0, width: 0, height: 0},
  insets: {top: 0, left: 0, right: 0, bottom: 0},
};

export const renderAppProviders = component => {
  return (
    <NativeBaseProvider initialWindowMetrics={inset}>
      {component}
    </NativeBaseProvider>
  );
};

export const mockState = (rootKey, state) => {
  return () => ({
    [rootKey]: state,
  });
};

export const mockDispatch = result => () => result;
