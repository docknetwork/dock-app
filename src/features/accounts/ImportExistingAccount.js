import React from 'react';
import {Box, Pressable, Stack} from 'native-base';
import {
  DocumentDownloadIcon,
  PlusCircleIcon,
  Text,
  OptionList,
  Theme,
} from 'src/design-system';
import {translate} from '../../locales';
import {BackIcon, Typography} from '../../design-system';

export function ImportExistingAccount({onSelect, onClose, onBack}) {
  return (
    <Stack p={8}>
      <Stack direction="row">
        <Pressable
          onPress={onBack}
          _pressed={{
            opacity: Theme.touchOpacity,
          }}>
          <Box pt={1} pr={5}>
            <BackIcon />
          </Box>
        </Pressable>
        <Typography variant="h1">
          {translate('import_account_modal.title')}
        </Typography>
      </Stack>
      <OptionList
        mt={5}
        postPress={onClose}
        items={[
          {
            title: translate('import_account_modal.recovery_phrase_option'),
            icon: <PlusCircleIcon />,
            onPress: () => onSelect('mnemonic'),
          },
          {
            title: translate('import_account_modal.upload_json_option'),
            icon: <DocumentDownloadIcon />,
            onPress: () => onSelect('json'),
          },
          {
            title: translate('import_account_modal.scan_qr_code_option'),
            icon: <DocumentDownloadIcon />,
            onPress: () => onSelect('qrcode'),
          },
        ]}
      />
    </Stack>
  );
}
