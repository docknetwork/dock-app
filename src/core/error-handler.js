import { Stack } from 'native-base';
import React from 'react';
import { Typography } from 'src/design-system';
import { translate } from 'src/locales';

export class ErrorBoundary extends React.Component<any, any> {
   constructor(props: any) {
      super(props);
      this.state = { hasError: false };
   }

   static getDerivedStateFromError(error: any) {
      return { hasError: true };
   }

   componentDidCatch(error: any, errorInfo: any) {
      // TODO: log the error in sentry
      console.error(error);
   }

   render() {
      if (this.state.hasError) {
         return <Stack><Typography>{translate('global.rendering_error')}</Typography></Stack>;
      }

      return this.props.children;
   }
}

export const withErrorBoundary = (WrapperComponent: any) => (props: any) => {
   return (
      <ErrorBoundary>
         <WrapperComponent {...props} />
      </ErrorBoundary>
   );
};
