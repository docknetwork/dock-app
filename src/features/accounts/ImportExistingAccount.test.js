import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {ImportExistingAccount} from './ImportExistingAccount';
import {renderAppProviders} from '../../core/test-utils';

const mockStore = configureMockStore();

describe('ImportExistingAccount', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(
      renderAppProviders(
        <ImportExistingAccount onClose={jest.fn()} visible={true} />,
      ),
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
