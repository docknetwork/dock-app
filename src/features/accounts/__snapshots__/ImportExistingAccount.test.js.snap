// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportExistingAccount should render correctly 1`] = `
<NativeBaseProvider
  initialWindowMetrics={
    Object {
      "frame": Object {
        "height": 0,
        "width": 0,
        "x": 0,
        "y": 0,
      },
      "insets": Object {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      },
    }
  }
>
  <ImportExistingAccount
    onClose={[MockFunction]}
    visible={true}
  />
</NativeBaseProvider>
`;
